controllers/vspherevm_controller.go

```diff
@@ -341,7 +346,14 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	}
 
 	// Update the VSphereVM's network status.
-	r.reconcileNetwork(ctx, vm)
+	if ok, err := r.reconcileNetwork(ctx, vm); !ok {
```

Comment:
Once you change the signature as requested below, this can become `if !r.reconcileNetwork(ctx, vm) {`

Reply:
Ack